src/addrman.cpp

```diff
@@ -7,20 +7,27 @@
 
 #include <hash.h>
 #include <serialize.h>
+#include <logging.h>
 
-int CAddrInfo::GetTriedBucket(const uint256& nKey) const
+int CAddrInfo::GetTriedBucket(const uint256& nKey, const std::vector<bool> &asmap) const
 {
     uint64_t hash1 = (CHashWriter(SER_GETHASH, 0) << nKey << GetKey()).GetCheapHash();
-    uint64_t hash2 = (CHashWriter(SER_GETHASH, 0) << nKey << GetGroup() << (hash1 % ADDRMAN_TRIED_BUCKETS_PER_GROUP)).GetCheapHash();
-    return hash2 % ADDRMAN_TRIED_BUCKET_COUNT;
+    uint64_t hash2 = (CHashWriter(SER_GETHASH, 0) << nKey << GetGroup(asmap) << (hash1 % ADDRMAN_TRIED_BUCKETS_PER_GROUP)).GetCheapHash();
+    int tried_bucket = hash2 % ADDRMAN_TRIED_BUCKET_COUNT;
+    uint32_t critical_as = GetCriticalAS(asmap);
+    LogPrint(BCLog::NET, "IP %s with critical AS%i belongs to tried bucket %i.\n", ToStringIP(), critical_as, tried_bucket);
+    return tried_bucket;
 }
 
-int CAddrInfo::GetNewBucket(const uint256& nKey, const CNetAddr& src) const
+int CAddrInfo::GetNewBucket(const uint256& nKey, const CNetAddr& src, const std::vector<bool> &asmap) const
 {
-    std::vector<unsigned char> vchSourceGroupKey = src.GetGroup();
-    uint64_t hash1 = (CHashWriter(SER_GETHASH, 0) << nKey << GetGroup() << vchSourceGroupKey).GetCheapHash();
+    std::vector<unsigned char> vchSourceGroupKey = src.GetGroup(asmap);
+    uint64_t hash1 = (CHashWriter(SER_GETHASH, 0) << nKey << GetGroup(asmap) << vchSourceGroupKey).GetCheapHash();
     uint64_t hash2 = (CHashWriter(SER_GETHASH, 0) << nKey << vchSourceGroupKey << (hash1 % ADDRMAN_NEW_BUCKETS_PER_SOURCE_GROUP)).GetCheapHash();
-    return hash2 % ADDRMAN_NEW_BUCKET_COUNT;
+    int new_bucket = hash2 % ADDRMAN_NEW_BUCKET_COUNT;
+    uint32_t critical_as = GetCriticalAS(asmap);
+    LogPrint(BCLog::NET, "IP %s with critical AS%i belongs to new bucket %i.\n", ToStringIP(), critical_as, new_bucket);
```

Comment:
> Okay I'm less happy with the last tiny logging commit [1619fe1](https://github.com/bitcoin/bitcoin/commit/1619fe188b278b82e9e147ee2de69abc4642c510) 2 reasons:

Agreed, refactoring this post-merge as a follow-up may be worthwhile.
 
> This is because you're not specifying `-debug=net`.

Thanks! I overlooked the `LogPrint(BCLog::NET)`.

nit: perhaps add a space between "AS" and the value in GetTriedBucket and GetNewBucket
```
2019-12-25T15:48:12Z IP 87.166.109.213 with critical AS3320 belongs to new bucket 238.
2019-12-25T15:48:12Z IP 84.44.193.228 with critical AS8422 belongs to new bucket 238.
2019-12-25T15:48:12Z IP 64.225.33.255 with critical AS200 belongs to new bucket 219.
2019-12-25T15:48:12Z IP 186.52.76.207 with critical AS6057 belongs to new bucket 301.
2019-12-25T15:48:12Z IP 186.4.77.132 with critical AS27964 belongs to new bucket 361.
2019-12-25T15:48:12Z IP 109.195.227.7 with critical AS57378 belongs to new bucket 808.
```